internal/provisioner/plain/controllers/bundle_controller.go

```diff
@@ -392,3 +393,68 @@ func (r *BundleReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		Owns(&corev1.ConfigMap{}).
 		Complete(r)
 }
+
+func (r *BundleReconciler) bundleImagePod(pod *corev1.Pod, source rukpakv1alpha1.ImageSource) *corev1.Pod {
+	if len(pod.Spec.InitContainers) != 1 {
+		pod.Spec.InitContainers = make([]corev1.Container, 1)
+	}
+
+	pod = r.addUnpackerInitContainer(pod)
+
+	if len(pod.Spec.Containers) != 1 {
+		pod.Spec.Containers = make([]corev1.Container, 1)
+	}
+
+	pod.Spec.Containers[0].Name = bundleUnpackContainerName
+	pod.Spec.Containers[0].Image = source.Ref
+	pod.Spec.Containers[0].Command = []string{"/bin/unpack", "--bundle-dir", "/"}
+	pod.Spec.Containers[0].VolumeMounts = []corev1.VolumeMount{{Name: "util", MountPath: "/bin"}}
+
+	return pod
+}
+
+func (r *BundleReconciler) bundleGitRepoPod(pod *corev1.Pod, source rukpakv1alpha1.GitSource) (*corev1.Pod, error) {
+	if len(pod.Spec.InitContainers) != 2 {
+		pod.Spec.InitContainers = make([]corev1.Container, 2)
+	}
+
+	pod = r.addUnpackerInitContainer(pod)
+
+	// Note: initContainer so we can ensure the repository has been cloned
+	// at the bundle.Spec.Source.Git.Ref before we unpack the Bundle contents
+	// that are stored in the repository.
+	pod.Spec.InitContainers[1].Name = "clone-repository"
+	// r.GitClientImage configures which git-based container image to use to clone the provided repository
+	// r.GitClientImage currently defaults to alpine/git:v2.32.0
+	pod.Spec.InitContainers[1].Image = r.GitClientImage
+	pod.Spec.InitContainers[1].ImagePullPolicy = corev1.PullIfNotPresent
+	cmd, err := git.CloneCommandFor(source)
+	if err != nil {
+		return nil, err
+	}
+	pod.Spec.InitContainers[1].Command = []string{"/bin/sh", "-c", cmd}
+	pod.Spec.InitContainers[1].VolumeMounts = []corev1.VolumeMount{{Name: "util", MountPath: "/util"}, {Name: "manifests", MountPath: "/manifests"}}
+
+	if len(pod.Spec.Containers) != 1 {
+		pod.Spec.Containers = make([]corev1.Container, 1)
+	}
+
+	pod.Spec.Containers[0].Name = bundleUnpackContainerName
+	pod.Spec.Containers[0].Image = r.GitClientImage
```

Comment:
Do we need to use the r.GitClientImage as the container image here? We should already have access to the /manifests directory as we populated that volume in an initContainer, and the r.UnpackImage should be smaller in size compared to any git container image we provide to this provisioner.